src/framework/MOM_io_file.F90

```diff
@@ -1729,35 +1729,41 @@ subroutine get_field_nc(handle, label, values, rescale)
 
   field_nc = handle%fields%get(label)
 
-  values(:,:) = 0.0
-  if (data_domain) then
-    allocate(values_nc(isc:iec,jsc:jec), source=0.)
-    iscl = isc+1-isd ; iecl = iec+1-isd ; jscl = jsc+1-jsd ; jecl = jec+1-jsd
-  else
-    iscl = 1 ; iecl = iec+1-isc ; jscl = 1 ; jecl = jec+1-jsc
-  endif
+  if (data_domain) &
+    allocate(values_c(1:iec-isc+1,1:jec-jsc+1))
```

Comment:
Because `values` is `intent(out)`, won't removing the `values(:,:) = 0.0` statement that was here before any of these changes lead to uninitialized values (and perhaps NaNs) appearing it its halo regions when data_domain is true, even if the incoming argument is properly initialized outside of this routine?  (In practice this might be fine with some compilers if they treat intent(out) like intent(inout).)  If so, even a halo update may not be able to eliminate these uninitialized variables.

Were `values` `intent(inout)`, I would agree that removing its initialization statement should be benign, if possibly outside of the scope of what is strictly necessary in this bug-fix PR.